apm-sniffer/apm-sdk-plugin/spring-plugins/core-patch/src/main/java/org/skywalking/apm/plugin/spring/patch/AutowiredAnnotationProcessorInterceptor.java

```diff
@@ -0,0 +1,110 @@
+ */
+
+package org.skywalking.apm.plugin.spring.patch;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+/**
+ * {@link AutowiredAnnotationProcessorInterceptor} return the correct constructor when the bean class is enhanced by
+ * skywalking.
+ *
+ * @author zhangxin
+ */
+public class AutowiredAnnotationProcessorInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+        MethodInterceptResult result) throws Throwable {
+
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+        Object ret) throws Throwable {
+        Class<?> beanClass = (Class<?>)allArguments[0];
+        if (EnhancedInstance.class.isAssignableFrom(beanClass)) {
+            Map<Class<?>, Constructor<?>[]> candidateConstructorsCache = (Map<Class<?>, Constructor<?>[]>)objInst.getSkyWalkingDynamicField();
+
+            Constructor<?>[] candidateConstructors = candidateConstructorsCache.get(beanClass);
+            if (candidateConstructors == null) {
+                Constructor<?>[] returnCandidateConstructors = (Constructor<?>[])ret;
+
+                /**
+                 * The return for the method {@link org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor#determineCandidateConstructors(Class, String)
+                 * contains three cases:
+                 * 1. Constructors with annotation {@link org.springframework.beans.factory.annotation.Autowired}
+                 * 2. The bean class only has one constructor with parameters
+                 * 3. The bean has constructor without parameters
+                 *
+                 * The interceptor does the following actions for the above three cases:
+                 * Case 1: Do nothing
+                 * Case 2: All the class that constrcutor enhance by skywalking will cannot in this case. because the manipulate mechanism generates another private constructor
+                 * in the enhance class.
+                 * Case 3: The interceptor fill out the private constructor when the class is enhanced by skywalking, and check if the remainder constructors size is equals one,
+                 * if yes, return the constructor. or return constructor without parameters.
```

Comment:
I can understand your points reluctantly, but I suggest you put the cases in this way:

1. What happened in each case after skywalking did the instrument by byte buddy?  Not influence or do the harm in which way.
1. What are the consequences?  
1. What are we going to do in this patch in order to fix?